storage_test.go

```diff
@@ -0,0 +1,199 @@
+
+	op1 := operation.Put([]byte("key1"), []byte("value1"))
+	op2 := operation.Delete([]byte("key2"))
+
+	result := txInstance.Then(op1, op2)
+
+	assert.Equal(t, txInstance, result, "Then should return the same tx instance")
+}
+
+func TestTxBuilder_Else(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	mockDriver := mocks.NewDriverMock(t)
+	strg := storage.NewStorage(mockDriver)
+	txInstance := strg.Tx(ctx)
+
+	op1 := operation.Put([]byte("key1"), []byte("value1"))
+	op2 := operation.Delete([]byte("key2"))
+
+	result := txInstance.Else(op1, op2)
+
+	assert.Equal(t, txInstance, result, "Else should return the same tx instance")
+}
+
+func TestTxBuilder_Commit_Success(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	mockDriver := mocks.NewDriverMock(t)
+	strg := storage.NewStorage(mockDriver)
+	txInstance := strg.Tx(ctx)
+
+	pred := predicate.ValueEqual([]byte("key"), "value")
+	thenOp := operation.Put([]byte("key"), []byte("new-value"))
+	elseOp := operation.Delete([]byte("key"))
+
+	txInstance.If(pred).Then(thenOp).Else(elseOp)
+
+	expectedResponse := tx.Response{
+		Succeeded: true,
+		Results:   []tx.RequestResponse{},
+	}
+
+	mockDriver.ExecuteMock.Expect(ctx, []predicate.Predicate{pred},
+		[]operation.Operation{thenOp}, []operation.Operation{elseOp}).
+		Return(expectedResponse, nil)
+
+	resp, err := txInstance.Commit()
+
+	require.NoError(t, err)
+	assert.Equal(t, expectedResponse, resp)
+	mockDriver.MinimockFinish()
+}
+
+func TestTxBuilder_Commit_Error(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	mockDriver := mocks.NewDriverMock(t)
+	strg := storage.NewStorage(mockDriver)
+	txInstance := strg.Tx(ctx)
+
+	pred := predicate.ValueEqual([]byte("key"), "value")
+	thenOp := operation.Put([]byte("key"), []byte("new-value"))
+
+	txInstance.If(pred).Then(thenOp)
+
+	expectedError := errors.New("driver execution failed") //nolint:err113 // Test error is fine
+
+	mockDriver.ExecuteMock.Expect(ctx, []predicate.Predicate{pred},
+		[]operation.Operation{thenOp}, []operation.Operation{}).
+		Return(tx.Response{Succeeded: false, Results: []tx.RequestResponse{}}, expectedError)
+
+	resp, err := txInstance.Commit()
+
+	require.Error(t, err)
+	require.EqualError(t, err, "tx execute failed: driver execution failed")
+
+	assert.False(t, resp.Succeeded)
+	assert.Nil(t, resp.Results)
+	mockDriver.MinimockFinish()
+}
+
+func TestTxBuilder_Chaining(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	mockDriver := mocks.NewDriverMock(t)
+	strg := storage.NewStorage(mockDriver)
+	txInstance := strg.Tx(ctx)
+
+	pred1 := predicate.ValueEqual([]byte("key1"), "value1")
+	pred2 := predicate.VersionEqual([]byte("key2"), int64(42))
+	thenOp1 := operation.Put([]byte("key1"), []byte("new-value1"))
+	thenOp2 := operation.Put([]byte("key2"), []byte("new-value2"))
+	elseOp := operation.Delete([]byte("key1"))
+
+	result := txInstance.
+		If(pred1).
+		Then(thenOp1).
+		Else(elseOp).
+		If(pred2).
```

Comment:
This usage looks quite confusing. Is there any practical meaning? As far as I see it is equal to:
```go
txInstance.
		If(pred1, pred2).
		Then(thenOp1, thenOp2).
		Else(elseOp)
```
which looks simpler.
Txn implementation of etcd client only allows straightforward flow if-then-else-commit with optional `else` or both `if` and `else` step and will panic with such a sequence so I suggest follow the same approach.